core/src/main/java/org/infinispan/protostream/impl/json/JsonTokenWriter.java

```diff
@@ -0,0 +1,75 @@
+package org.infinispan.protostream.impl.json;
+
+import static org.infinispan.protostream.impl.json.JsonHelper.NULL;
+
+/**
+ * Writer for a JSON token.
+ *
+ * <p>
+ * These writers are utilized when transforming the {@link JsonToken} into the final string representation. Most of the
+ * tokens in the JSON use a single character, except <code>STRING</code> and the value. A <code>STRING</code> represent
+ * the property/key in the JSON, and the value is the actual value.
+ * </p>
+ */
+interface JsonTokenWriter {
+
+   /**
+    * Writes the token to the given output.
+    *
+    * @param out The output to write the token.
+    */
+   void write(StringBuilder out);
```

Comment:
Is it worth making this a higher level interface, e.g. `Appendable`?

I can't think of a particular use-case, I'm just thinking it might provide more flexibility in the future.

Reply:
I can make that work. I'll create the `Appendable` outside and extend the interface here.